src/NuGet.Indexing/VerifiedPackages.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using FrameworkLogger = Microsoft.Extensions.Logging.ILogger;
+using System.IO;
+
+namespace NuGet.Indexing
+{
+    public static class VerifiedPackages
+    {
+        public static HashSet<string> Load(string fileName, ILoader loader, FrameworkLogger logger)
+        {
+            try
+            {
+                using (var reader = loader.GetReader(fileName))
+                {
+                    return Parse(reader);
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("Unable to load {FileName} as deserialization threw: {Exception}", fileName, ex);
+
+                throw;
+            }
+        }
+
+        public static HashSet<string> Parse(JsonReader reader)
+        {
+            // The file should contain an array of strings, such as: ["Package1","Package2", ...]
+            reader.Read();
+            ThrowIfNotExpectedToken(reader, JsonToken.StartArray);
+
+            // Read all of the package ID strings from the array.
+            string packageId = reader.ReadAsString();
+            var result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            while (packageId != null)
+            {
+                result.Add(String.Intern(packageId));
```

Comment:
Assuming String.Intern is intended to optimize lookup performance? Would the string for lookup need to be interned as well, and would the optimization work on a case insensitive collection?

Would measure if this code actually improves perf (i.e., with simple console app), and keep / add comment if it does.